src/backend/executor/nodeHash.c

```diff
@@ -2192,7 +2192,12 @@ bool
 ExecScanHashBucket(HashState *hashState, HashJoinState *hjstate,
 				   ExprContext *econtext)
 {
-	ExprState  *hjclauses = hjstate->hashclauses;
```

Comment:
how about set `hashclauses` to NULL ?

Reply:
I don't try it, in pg, it remains not NULL, just follow pg here.